### DISCUSSION TITLE
remains of a comma operator?

### DISCUSSION THREAD

**Discussion by esakilpi** [hylang/hy#2288] [Q&A] [answered]
This is from an example in Mark Watsons book "A LIsp Programmer Living in Python-Land" (page 52). Variable all-data is a pandas DataFrame that has been read from a csv file.

`(setv x-data10 (. all-data.iloc [(, (slice 0 10) [0 1 2 3 4 5 6 7 8])] values))`

I cannot get it working. The problem is the comma before "slice". The error message is "NameError: name 'hyx_XcommaX' is not defined".

It looks like a comma operator, but has there ever been one in hy? I have seen some old discussion mentioning it. The problem is that I cannot figure out what the comma is doing there and how to change the code to make it do what it is supposed to do,

**Comment by allison-casey** [comment, marked as answer]
Python tuples were given their own dedicated syntax like all other datastructures instead of piggybacking off the comma expression syntax. i.e. `#(1 2 3)`

 All the following are equivalent
```clojure
(setv x-data10 (. all-data.iloc [#((slice 0 10) [0 1 2 3 4 5 6 7 8])] values))
(setv x-data10 (.values (get all-data.iloc #((slice 0 10) [0 1 2 3 4 5 6 7 8]))))
```
there's also [`ncut`](https://hyrule.readthedocs.io/en/master/#hyrule.collections.ncut) from [hyrule](https://github.com/hylang/hyrule), which helps with this kind of multi-dimensional slicing 
```clojure
(setv x-data10 (.values (ncut all-data.iloc 0:10 [0 1 2 3 4 5 6 7 8])))
```

**Reply by esakilpi** [reply]
Thanks!
